cpp/dolfinx/fem/interpolate.h

```diff
@@ -357,33 +525,70 @@ void interpolate_nonmatching_maps(Function<T>& u1, const Function<T>& u0,
         array1[dof_bs1 * dofs1[i] + k] = local1[dof_bs1 * i + k];
   }
 }
-} // namespace impl
+//----------------------------------------------------------------------------
+/// Interpolate from one finite element Function to another
+/// @param[out] u The function to interpolate into
+/// @param[in] v The function to be interpolated
+/// @param[in] cells List of cell indices to interpolate on
+template <typename T>
+void interpolate_nonmatching_meshes(Function<T>& u, const Function<T>& v,
+                                    const xtl::span<const std::int32_t>& cells)
+{
+  assert(u.function_space());
+  assert(v.function_space());
 
-/// Compute the evaluation points in the physical space at which an
-/// expression should be computed to interpolate it in a finite elemenet
-/// space.
-///
-/// @param[in] element The element to be interpolated into
-/// @param[in] mesh The domain
-/// @param[in] cells Indices of the cells in the mesh to compute
-/// interpolation coordinates for
-/// @return The coordinates in the physical space at which to evaluate
-/// an expression. The shape is (3, num_points) and storage is row-major.
-std::vector<double>
-interpolation_coords(const FiniteElement& element, const mesh::Mesh& mesh,
-                     const xtl::span<const std::int32_t>& cells);
+  std::shared_ptr<const mesh::Mesh> mesh = u.function_space()->mesh();
+  std::shared_ptr<const mesh::Mesh> mesh_v = v.function_space()->mesh();
+
+  assert(mesh);
+
+  int result;
```

Comment:
Use scoping to avoid leaking variables this are used very locally from entering the scope of the whole function.